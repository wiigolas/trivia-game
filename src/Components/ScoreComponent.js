import React, { Component } from 'react';
import { View, Text } from 'react-native';
import styles from '../Styles/styles.js';

export default class ScoreComponent extends Component<Props> {
  constructor() {
    super();
    this.state = {};
  }

  scoreFeedback = (totalScore) => {
    let message;

    switch (totalScore) {
      case 0:
        message = 'You suck!';
        break;
      case 1:
      case 2:
      case 3:
        message = 'That was pretty bad!';
        break;
      case 4:
      case 5:
      case 6:
        message = 'Well done, keep on trying!';
        break;
      case 7:
      case 8:
      case 9:
        message = 'You\'re really smart!';
        break;
      case 10:
        message = 'You\'re a genius! Are you an alien? 10/10';
        break;
    }

    return message;
  }

  render() {
    const totalScore = `You had ${this.props.totalScore} correct answers`;
    return(
      <View style={styles.container}>
        <Text id='score'>{totalScore}</Text>
        <Text id='feedback'>{this.scoreFeedback(this.props.totalScore)}</Text>
      </View>
    )
  }
}
