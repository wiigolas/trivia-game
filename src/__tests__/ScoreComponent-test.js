import 'react-native';
import React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import sinon from 'sinon';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep'}));

configure({ adapter: new Adapter() });

import ScoreComponent from '../Components/ScoreComponent';

describe('testing the <ScoreComponent>', () => {
  let wrapper;

  beforeEach(() => {
    this.wrapper = shallow(
      <ScoreComponent totalScore={5}/>
    );
  })
  
  it('renders correctly', () => {
    expect(this.wrapper).toMatchSnapshot();
  });

  it('displays score feedback', () => {
    expect(this.wrapper.find('Text').first().props().children).toEqual('You had 5 correct answers');
    expect(this.wrapper.find('Text').last().props().children).toEqual('Well done, keep on trying!');
  })
})
