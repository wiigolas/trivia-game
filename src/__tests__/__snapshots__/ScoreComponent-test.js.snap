// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the <ScoreComponent> renders correctly 1`] = `
<Component>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    id="score"
  >
    You had 5 correct answers
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    id="feedback"
  >
    Well done, keep on trying!
  </Text>
</Component>
`;
